metadetect/lsst_measure_scarlet.py

```diff
@@ -0,0 +1,992 @@
+
+    Parameters
+    ----------
+    wcs: stack wcs
+        The wcs for the full image not this sub image
+    imobj: lsst.afw.image.Exposure
+        An Exposure object, e.g. ExposureF
+    source: lsst.afw.table.SourceRecord
+        A source record from detection/deblending
+
+    returns
+    --------
+    obs: ngmix.Observation
+        The Observation unless all the weight are zero, in which
+        case None is returned
+    """
+
+    im = subim.image.array
+
+    wt = _extract_weight(subim)
+    if np.all(wt <= 0):
+        return None
+
+    maskobj = subim.mask
+    bmask = maskobj.array
+
+    fp = source.getFootprint()
+    peak = fp.getPeaks()[0]
+
+    # this is a Point2D but at an integer location
+    peak_location = peak.getCentroid()
+
+    jacob = _extract_jacobian(
+        wcs=wcs,
+        subim=subim,
+        source=source,
+        orig_cen=peak_location,
+    )
+
+    psf_im = _extract_psf_image(exposure=subim, orig_cen=peak_location)
+
+    # fake the psf pixel noise
+    psf_err = psf_im.max()*0.0001
+    psf_wt = psf_im*0 + 1.0/psf_err**2
+
+    # use canonical center for the psf
+    psf_cen = (np.array(psf_im.shape)-1.0)/2.0
+    psf_jacob = jacob.copy()
+    psf_jacob.set_cen(row=psf_cen[0], col=psf_cen[1])
+
+    # we will have need of the bit names which we can only
+    # get from the mask object
+    # this is sort of monkey patching, but I'm not sure of
+    # a better solution
+    meta = {'maskobj': maskobj}
+
+    psf_obs = ngmix.Observation(
+        psf_im,
+        weight=psf_wt,
+        jacobian=psf_jacob,
+    )
+    obs = ngmix.Observation(
+        im,
+        weight=wt,
+        bmask=bmask,
+        jacobian=jacob,
+        psf=psf_obs,
+        meta=meta,
+    )
+
+    return obs
+
+
+def _extract_jacobian(wcs, subim, source, orig_cen):
+    """
+    extract an ngmix.Jacobian from the image object
+    and object record
+
+    Parameters
+    ----------
+    wcs: stack wcs
+        The wcs for the full image not this sub image
+    imobj: lsst.afw.image.Exposure
+        An Exposure object, e.g. ExposureF
+    source: lsst.afw.table.SourceRecord
+        A source record from detection/deblending
+    orig_cen: Point2D
+        Location of object in original image
+
+    returns
+    --------
+    Jacobian: ngmix.Jacobian
+        The local jacobian
+    """
+
+    xy0 = subim.getXY0()
+    stamp_cen = orig_cen - geom.Extent2D(xy0)
+
+    LOG.debug('cen in subim: %s', stamp_cen)
+    row = stamp_cen.getY()
+    col = stamp_cen.getX()
+
+    # we get this at the original center
+    linear_wcs = wcs.linearizePixelToSky(
+        orig_cen,  # loc in original image
+        geom.arcseconds,
+    )
+    jmatrix = linear_wcs.getLinear().getMatrix()
+
+    jacob = ngmix.Jacobian(
+        row=row,
+        col=col,
+        dudcol=jmatrix[0, 0],
+        dudrow=jmatrix[0, 1],
+        dvdcol=jmatrix[1, 0],
+        dvdrow=jmatrix[1, 1],
+    )
+
+    return jacob
+
+
+def _extract_weight(subim):
```

Comment:
Why take a median here instead of using 1/var of input where that is valid?

Reply:
For weighed moments we can't have any zero weight points or we will get a get a bias. It is plausible to me that metacal can't calibrate that, but I haven't tested this.

Reply:
For weighted moments you can do ok if you symmetrize the zero weight pixels in the mask. I tested this. 4-fold symmetry is usually enough.

Reply:
I'm gonna make an issue to just turn that on all the time maybe.